analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/SymbolicValueTest.cs

```diff
@@ -158,5 +159,34 @@ public void Equals_ReturnsTrueForSameConstraints()
             basicSingle.Equals((object)null).Should().BeFalse();
             basicSingle.Equals((SymbolicValue)null).Should().BeFalse();     // Explicit cast to ensure correct overload
         }
+
+        [TestMethod]
+        public void AllConstraints_Empty() =>
+            new SymbolicValue().AllConstraints().Should().BeEmpty();
+
+        [TestMethod]
+        public void AllConstraints_ResturnsConstraints()
+        {
+            var sut = new SymbolicValue().WithConstraint(TestConstraint.First).WithConstraint(DummyConstraint.Dummy);
+            sut.AllConstraints().Should().HaveCount(2).And.Contain(new SymbolicConstraint[] { TestConstraint.First, DummyConstraint.Dummy });
+        }
+
+        [TestMethod]
+        public void AllConstraints_ContainsOneConstraintPerConstriantType()
+        {
+            var sut = new SymbolicValue()
+                .WithConstraint(TestConstraint.First)
+                .WithConstraint(TestConstraint.Second)
+                .WithConstraint(DummyConstraint.Dummy)
+                .WithConstraint(DummyConstraint.Dummy);
+            sut.AllConstraints().Should().HaveCount(2).And.Contain(new SymbolicConstraint[] { TestConstraint.Second, DummyConstraint.Dummy });
+        }
+
+        [TestMethod]
+        public void AllConstraints_DontContainsRemovedConstraint()
+        {
+            var sut = new SymbolicValue().WithConstraint(TestConstraint.First).WithoutConstraint(TestConstraint.First);
+            sut.AllConstraints().Should().BeEmpty();
+        }
```

Comment:
This is already tested by tests for `WithConstraint` and `WithoutConstraint`. We can remove these for simplicity. The first two should be enough.
```suggestion
```